model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -147,6 +148,37 @@ class ModelReplicationServer(
             call.respondDelta(versionHash, baseVersionHash)
         }
 
+        delete<Paths.deleteRepositoryBranch> {
+            val repositoryName = call.parameters["repository"]
+            if (repositoryName == null) {
+                call.respondText("Request lacks repository name", status = HttpStatusCode.BadRequest)
+                return@delete
+            }
+            val repositoryId = try {
+                RepositoryId(repositoryName)
```

Comment:
Maybe add a method to `RepositoryId` to parse and optionally return an ID.
Not relying on the exception would also get rid of the Detekt warning.

Reply:
Also this case is not tested.

Reply:
I'm fine with the detect warning. That's a problem for all top-level exceptions that are handled differently from rethrowing them somehow. And the ktor approach requires this.

We have to generally refactor this for the whole file, similar to the exception handling. I'd like to postpone that to a separate PR.